docs/rfcs/rfc-040-relay-first-api.md

```diff
@@ -0,0 +1,1282 @@
+    title: String!
+    alternativeTitles: [String!]
+    released: Int
+    actors: [Person!]! @relationship(type: "ACTED_IN", direction: IN, properties: "ActedIn")
+    director: Person! @relationship(type: "DIRECTED", direction: IN)
+}
+
+type Person {
+    name: String!
+    movies: [Movie!]! @relationship(type: "ACTED_IN", direction: OUT, properties: "ActedIn")
+    directed: Movie @relationship(type: "DIRECTED", direction: OUT)
+}
+
+interface ActedIn @relationshipProperties {
+    year: Int
+}
+```
+
+## Connections API - Query
+
+Query will be done with the following operations:
+
+```graphql
+type Query {
+    moviesConnection(
+        first: Int
+        after: String
+        last: Int
+        before: String
+        where: MovieConnectionWhere
+        sort: [MovieConnectionSort]
+    ): MoviesConnection!
+    peopleConnection(
+        first: Int
+        after: String
+        where: PersonConnectionWhere
+        sort: [PersonConnectionSort]
+    ): PeopleConnection!
+}
+```
+
+The `*Connection` types will be a Relay compliant [Connection Type](https://relay.dev/graphql/connections.htm#sec-Connection-Types). With the following fields:
+
+```graphql
+type MoviesConnection {
+    pageInfo: PageInfo! # Required by relay
+    edges: [MovieEdge!]! # Required by relay
+    totalCount: Int!
+    aggregation: MoviesAggregation
+}
+```
+
+A simple query would look like:
+
+```graphql
+query MoviesTitles {
+    moviesConnection {
+        edges {
+            node {
+                title
+            }
+        }
+    }
+}
+```
+
+### Edges
+
+The `*Edge` types will be a Relay [Edge Type](https://relay.dev/graphql/connections.htm#sec-Edge-Types). These fields will be used to traverse the API and hold the relationship properties, if any:
+
+```graphql
+type MovieActorsEdge {
+    cursor: String! # Required by Relay
+    node: PersonNode! # Required by Relay
+    fields: ActedIn # Fields of the relationship
+}
+```
+
+There will be an Edge type per entity, as well as a different Edge type per relationship, relationship edges may contain the property `fields` that will hold the relationship properties.
+
+A query following an edge and its properties would look like:
+
+```graphql
+query MoviesWithActors {
+    moviesConnection {
+        edges {
+            node {
+                title
+                actors {
+                    edges {
+                        node {
+                            name
+                        }
+                        fields {
+                            year
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+In this example, we get the related Actor name, as well as the property `year` from the relationship.
+
+### Nodes
+
+The Entity (e.g. movies) properties for the projection will be defined in a `*Node` type:
+
+```graphql
+type MovieNode {
+    title: String!
+    released: Int
+    alternativeTitles: [String!]!
+    actors(
```

Comment:
I may be miss-remembering but did we not say this would be `actorsConnection` to keep consistent and because there some confusion over whether or not this was a connection or basic type

Reply:
yeah, but after some discussion we saw that where filtering would get extremely verbose so we didn't change it **yet**

We should get some examples and extra discussion, but it shouldn't be a blocker for the overall design anyway

Reply:
Yeah I agree this shouldn't be a blocker. I didn't really have a strong opinion on it either way, just wanted to make sure it wasn't forgotten!

Reply:
As you say, it might be interesting to see an example in an "alternatives considered" section

Reply:
That's a good point, I'll try to compile examples for when we have further discussions on these topics